优化finetune.py为混合精度训练.

mellivoraPKU opened the pull request:
1.优化finetune.py为混合精度训练.
2.将finetune_distributed.py分布式训练方式改为deepspeed,同时相应开启混合训练. 
3.更新requirements.txt到最新版本

zhangfaen commented on finetune.py:
```diff
@@ -230,7 +235,19 @@ def train():
-            outputs = model(**inputs, labels=labels)
+
+            # Automatic Mixed Precision (AMP) context manager for efficient training.
+            # Inside this context, `torch.is_autocast_enabled()` returns True; outside the context, it returns False.
```
# Inside this context, `torch.is_autocast_enabled()` returns True; outside the context, it returns False.

This line should be changed to below 
--------
AMP context manager can be embeded by another AMP context manager, the behavior is shown by below code snippet.

# import torch
# from torch.amp import autocast

# with autocast(device_type='cuda', dtype=torch.bfloat16):
   # print(torch.is_autocast_enabled()) # True

    # with autocast(device_type='cuda', dtype=torch.bfloat16):
      # print(torch.is_autocast_enabled()) # True

    # with autocast(device_type='cuda', dtype=torch.bfloat16, enabled=False):
      # print(torch.is_autocast_enabled()) # False
  
    # print(torch.is_autocast_enabled()) # True

mellivoraPKU committed:
1.优化finetune.py为混合精度训练.
2.将finetune_distributed.py分布式训练方式改为deepspeed,同时相应开启混合训练.
3.更新requirements.txt到最新版本
